packages/web-styles/src/components/stepper.scss

```diff
@@ -1,165 +1,139 @@
-@use '../mixins/size' as size-mixins;
+@use '../mixins/size' as size-mx;
 @use '../mixins/utilities';
 
-@use '../variables/icons';
-@use '../variables/color';
 @use '../variables/spacing';
-@use '../variables/type';
 @use '../variables/components/stepper';
 
 @use '../themes/bootstrap/overrides' as *;
 
-.stepper {
-  @include size-mixins.responsive-size('big', 'margin-bottom');
+.stepper-container {
+  @include size-mx.responsive-size('big', 'margin-bottom');
   position: relative;
-  padding: 0 stepper.$step-indicator-size * 0.5;
 }
 
 .stepper-bar {
-  height: stepper.$stepper-bar-size;
-  margin-top: stepper.$step-indicator-size * 0.5;
-  margin-bottom: -1 * (stepper.$stepper-bar-size + stepper.$step-indicator-size) * 0.5;
+  height: stepper.$stepper-bar-height;
+  margin-top: stepper.$stepper-indicator-height * 0.5;
+  margin-bottom: -1 * (stepper.$stepper-bar-height + stepper.$stepper-indicator-height) * 0.5;
 }
 
-.steps {
-  display: flex;
-  justify-content: space-between;
+.stepper {
+  counter-reset: step-index;
+
+  list-style: none;
+  padding: 0 (0.5 * stepper.$stepper-indicator-height);
+  margin: 0;
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(#{0.5 * stepper.$stepper-indicator-height}, 1fr));
+  justify-items: center;
 }
 
-.step {
+.stepper-item {
+  counter-increment: step-index;
   display: flex;
-  flex: 1 1 100%;
   flex-direction: column;
-  align-items: center;
-
-  &:first-child,
-  &:last-child {
-    flex-basis: 50%;
-  }
 
   &:first-child {
-    align-items: flex-start;
+    justify-self: start;
+    align-items: start;
+    transform: translateX(#{-0.5 * stepper.$stepper-indicator-height});
   }
 
   &:last-child {
-    align-items: flex-end;
-  }
-
-  &,
-  &:hover {
-    text-decoration: none;
+    justify-self: end;
+    align-items: end;
+    transform: translateX(#{0.5 * stepper.$stepper-indicator-height});
   }
-}
-
-.step-indicator {
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  width: stepper.$step-indicator-size;
-  height: stepper.$step-indicator-size;
-  margin: 0 auto;
-  border: spacing.$size-micro solid color.$white;
-  border-radius: 100%;
-  background-color: stepper.$step-indicator-default-bg;
-  color: stepper.$step-indicator-default-color;
-  font-weight: type.$font-weight-bold;
 
-  .step:first-child & {
-    margin-left: math.div(stepper.$step-indicator-size, -2);
+  &:not(:first-child):not(:last-child) {
+    grid-column-end: span 2;
+    align-items: center;
   }
 
-  .step:last-child & {
-    margin-right: math.div(stepper.$step-indicator-size, -2);
-  }
-
-  .step.current &,
-  .step.passed & {
-    background-color: stepper.$step-indicator-active-bg;
-    color: stepper.$step-indicator-active-color;
+  &[aria-current] {
+    pointer-events: none;
   }
+}
 
-  .step.passed & {
-    background-image: stepper.$step-indicator-passed-icon;
-    background-repeat: no-repeat;
+// Stepper indicator styles
+.stepper-item {
```

Comment:
Can you combine this selector with the `.stepper-item` above?

https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=308&id=swisspost_common-web-frontend&open=AYKNVsMzXxzs6oVwk4vE